src/GrasshopperTeklaDrawingLink/Components/Text/TextAttributesComponent.cs

```diff
@@ -0,0 +1,132 @@
+﻿using Grasshopper.Kernel;
+
+using GTDrawingLink.Tools;
+using GTDrawingLink.Types;
+
+using System.Collections.Generic;
+
+using Tekla.Structures.Drawing;
+
+using TSD = Tekla.Structures.Drawing;
+
+namespace GTDrawingLink.Components.Text
+{
+    public class TextAttributesComponent : TeklaComponentBase
+    {
+        public override GH_Exposure Exposure => GH_Exposure.primary;
+
+        public TextAttributesComponent()
+            : base(ComponentInfos.TextAttributesComponent)
+        {
+        }
+
+        protected override void RegisterInputParams(GH_InputParamManager pManager)
+        {
+            SetParametersAsOptional(pManager, new List<int> {
+                pManager.AddParameter(new FontAttributesParam(ParamInfos.FontAttributes, GH_ParamAccess.item)),
+                pManager.AddParameter(new EnumParam<FrameTypes>(ParamInfos.FrameType, GH_ParamAccess.item)),
+                pManager.AddParameter(new EnumParam<DrawingColors>(ParamInfos.DrawingColor, GH_ParamAccess.item)),
+                pManager.AddParameter(new ArrowAttributesParam(ParamInfos.ArrowAttribute, GH_ParamAccess.item)),
+                pManager.AddBooleanParameter(
```

Comment:
Ok, now I get why you have not used wrappers from TeklaComponentBase - they are not returning index of the parameter.

It will be non-breaking change if you modify the signature of all of the GH_InputParamManager Add...Parameter from the TeklaComponentBase to return int - index of freshly added parameter.

By doing this you could still use this nice wrapper  SetParametersAsOptional with preserving shortec declaration inside of RegisterInputParams